src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -229,55 +228,60 @@ private static void logStartupMessage(Class<? extends Name<?>>[] classes, Inject
       final String doc = annotation.doc();
       final String defaultVal;
       {
-        if (!annotation.default_value().equals(NamedParameter.REEF_UNINITIALIZED_VALUE))
+        if (!annotation.default_value().equals(NamedParameter.REEF_UNINITIALIZED_VALUE)) {
           defaultVal = annotation.default_value();
-        else if (!annotation.default_class().equals(Void.class))
+        } else if (!annotation.default_class().equals(Void.class)) {
           defaultVal = annotation.default_class().getSimpleName();
-        else if (annotation.default_values().length > 0)
+        } else if (annotation.default_values().length > 0) {
           defaultVal = Arrays.toString(annotation.default_values());
-        else if (annotation.default_classes().length > 0) {
-          Class<?>[] default_classes = annotation.default_classes();
-          String[] sss = new String[default_classes.length];
-          for (int i1 = 0; i1 < default_classes.length; i1++) {
-            sss[i1] = default_classes[i1].getSimpleName();
+        } else if (annotation.default_classes().length > 0) {
+          StringBuilder b = new StringBuilder("[");
+          Class<?>[] cls = annotation.default_classes();
+          for (int j = 0; ; j++) {
+
+            b.append(cls[j].getSimpleName());
+            if (j == cls.length - 1) {
+              b.append(']');
+              break;
+            }
+            b.append(", ");
           }
-          defaultVal = Arrays.toString(sss);
+          defaultVal = b.toString();
         } else defaultVal = "";
       }
 
-      if (i > 0) sb.append('\n');
+      if (i > 0) {
+        sb.append('\n');
+      }
       sb.append(simpleName);
-      if (!shortName.isEmpty()) sb.append(" [-").append(shortName).append("]");
+      if (!shortName.isEmpty()) {
+        sb.append(" [-").append(shortName).append("]");
+      }
 
-      //      sb.append(" (").append(expectedParameter.getSimpleArgName()).append("): ")
+      // If desired, the type of the parameter can be obtained from
+      // commandLineConf.getNamedParameters() --> .getSimpleArgName()
       sb.append(": ").append(doc).append("\n -> ");
 
       boolean ok;
       try {
-        if ((ok = inj.isParameterSet(fullName))) sb.append(inj.getInstance(fullName).toString());
+        ok = inj.isParameterSet(fullName);
+        if (ok) {
+          sb.append(inj.getInstance(fullName).toString());
+        }
       } catch (InjectionException e) {
```

Comment:
My guess is that the parameter could be set, but a bad parameter could be provided.  For example, a string "abc" could be provided where an int parameter is expected. Then the InjectionException might get thrown.  I can't tell by inspecting the source code; it's complicated.  Instead I am being proactively defensive.

Since you mentioned the BindException, I will add that in.